### PR TITLE
[RPD-193] Refactor `TemplateRunner.is_approved` into a UI function

### PR DESCRIPTION
**Pull request by swells2020**
This pull request refactors the `is_approved` logic, which is currently a method of the `runners.azure_runner.AzureRunner` class, into a separate module within the `cli.ui` submodule.  The logic of the function is unchanged; imports and references have been updated to reflect the new location.

## Checklist

Please ensure you have done the following:

* [X] I have read the [CONTRIBUTING](https://github.com/fuzzylabs/matcha/blob/main/CONTRIBUTING.md) guide.
* [ ] I have updated the documentation if required.
* [X] I have added tests which cover my changes.

## Type of change

Tick all those that apply:

* [ ] Bug Fix (non-breaking change, fixing an issue)
* [ ] New feature (non-breaking change to add functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [X] Other (add details above)

### CODE REVIEW

File: src/matcha_ml/cli/ui/functions.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+from matcha_ml.cli.ui.emojis import Emojis
+from matcha_ml.cli.ui.print_messages import (
+    print_status,
+)
+from matcha_ml.cli.ui.status_message_builders import (
+    build_resource_confirmation,
+)
+
+
+def is_approved(verb: str, resources: List[Tuple[str, str]]) -> bool:
```
[the remark this answers is not in the document]
Comment (Christopher-Norman) [reply src/matcha_ml/cli/ui/functions.py:15]: `get_modify_resource_aproval`

**Review by Christopher-Norman — CHANGES_REQUESTED** [review]
Looking good, just one question about possibly updating a function name 👍

File: src/matcha_ml/cli/ui/functions.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (swells2020) [reply src/matcha_ml/cli/ui/functions.py]: Updated, thanks Chris!

File: tests/test_cli/test_ui_functions.py [hunk 2]
```diff
@@ -2,14 +2,14 @@
 
 from unittest import mock
 
-from matcha_ml.cli.ui.functions import is_approved
+from matcha_ml.cli.ui.functions import get_modify_resource_approval
 
 
-def test_is_approved():
+def test_get_modify_resource_approvals():
     """Test if is_approved behaves as expected based on user's input."""
```
Comment (Christopher-Norman) [inline tests/test_cli/test_ui_functions.py:9]: Update docstring with new function name

Comment (swells2020) [reply tests/test_cli/test_ui_functions.py]: Thanks for the spot!

File: src/matcha_ml/cli/ui/functions.py [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+from matcha_ml.cli.ui.emojis import Emojis
+from matcha_ml.cli.ui.print_messages import (
+    print_status,
+)
+from matcha_ml.cli.ui.status_message_builders import (
+    build_resource_confirmation,
+)
+
+
+def get_modify_resource_approval(verb: str, resources: List[Tuple[str, str]]) -> bool:
```
Comment (JonoCX) [inline src/matcha_ml/cli/ui/functions.py:15]: For functions that return a bool, this function name is slightly more descriptive: `is_user_approved()`

File: src/matcha_ml/cli/ui/functions.py [hunk 4]
```diff
@@ -0,0 +1,32 @@
+"""UI functions."""
```
Comment (JonoCX) [inline src/matcha_ml/cli/ui/functions.py:1]: Nit: Having a `functions.py` feels slightly too ambiguous - could this be renamed to something more descriptive?

Nit: "UI functions" doesn't really tell me much, what type of functions can be found here?

File: tests/test_cli/test_ui_functions.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
+"""Tests for cli.ui.functions."""
+
+from unittest import mock
+
+from matcha_ml.cli.ui.functions import get_modify_resource_approval
+
+
+def test_get_modify_resource_approvals():
```
Comment (JonoCX) [inline tests/test_cli/test_ui_functions.py:8]: This test current tests two different conditions, one where the user gives their approval and another where the user doesn't. A unit test should test a single, discrete condition - these should be split into two separate tests.

**Review by JonoCX — CHANGES_REQUESTED** [review]
Nice work Callum, this is looking cleaner. See comments

File: src/matcha_ml/cli/ui/functions.py (hunk 3 above)
Comment (swells2020) [reply src/matcha_ml/cli/ui/functions.py:15]: Updated, thanks Jon

File: src/matcha_ml/cli/ui/functions.py (hunk 4 above)
Comment (swells2020) [reply src/matcha_ml/cli/ui/functions.py:1]: Updated the docstring and filenames to reflect that the module is specifically for user approval functions.

File: tests/test_cli/test_ui_functions.py (hunk 5 above)
Comment (swells2020) [reply tests/test_cli/test_ui_functions.py]: Thanks Jon, they're split.

File: tests/test_cli/test_ui_functions.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+
+def test_user_approves():
+    """Test if is_user_approved behaves as expected where user provides approval."""
+    with mock.patch("typer.confirm") as mock_confirm:
+        mock_confirm.return_value = True
+        assert is_user_approved("provision", [])
+
+
+def test_user_does_approves():
```
Comment (JonoCX) [inline tests/test_cli/test_ui_functions.py:15]: `test_user_does_approve` -> `test_user_does_not_approve`?

**Review by JonoCX — CHANGES_REQUESTED** [review]
Looking godd! One minor change required to the name of the test function

**Pull request merged**